packages/pigeon/lib/objc_generator.dart

```diff
@@ -232,52 +234,91 @@ NSObject<FlutterMessageCodec>* ${_getCodecGetterName(options.prefix, api.name)}(
 ''');
 }
 
+String _spaceJoin(String x, String y) => '$x $y';
+
+String _capitalize(String str) =>
+    (str.isEmpty) ? '' : str[0].toUpperCase() + str.substring(1);
+
+Iterable<String> _getSelectorComponents(Method func) sync* {
+  final Iterator<NamedType> it = func.arguments.iterator;
+  final bool hasArguments = it.moveNext();
+  yield '${func.name}${hasArguments ? _capitalize(func.arguments[0].name) : ''}';
+  while (it.moveNext()) {
+    yield it.current.name;
+  }
+}
+
+String _makeObjcSignature({
+  required Method func,
+  required ObjcOptions options,
+  required String returnType,
+  required String lastArgType,
+  required String lastArgName,
+  String Function(int, NamedType)? argNameFunc,
+}) {
+  argNameFunc = argNameFunc ?? (int _, NamedType e) => e.name;
+  final Iterable<String> argNames =
+      followedByOne(indexMap(func.arguments, argNameFunc), lastArgName);
+  final Iterable<String> selectorComponents = ((Iterable<String> x) =>
+          func.arguments.isEmpty ? x : followedByOne(x, lastArgName))(
+      _getSelectorComponents(func));
+  final Iterable<String> argTypes = followedByOne(
+    func.arguments.map((NamedType arg) {
+      final String nullable = func.isAsynchronous ? 'nullable ' : '';
+      final String argType = _objcTypeForDartType(options.prefix, arg.type);
+      return '$nullable$argType *';
+    }),
+    lastArgType,
+  );
+
+  final String argSignature = map3(
+    selectorComponents,
+    argTypes,
+    argNames,
+    (String component, String argType, String argName) =>
+        '$component:($argType)$argName',
+  ).join(' ');
+  return '-($returnType)$argSignature';
```

Comment:
We should follow Google style; space between `-` and `(`.

Reply:
Done